src/compute_graph/compute_graph.jl

```diff
@@ -0,0 +1,98 @@
+include("tensor_cache.jl")
+
+using AbstractTrees
+using QXContexts
+
+export build_compute_graph
+
+"""
+build_compute_graph(tnc::TensorNetworkCircuit,
+                   plan::Vector{<:Tuple}),
+                   bond_groups::Union{Nothing, Vector{<:Vector{<:Index}}}=nothing)
+
+Function to build a compute tree from a tensor network and plan
+"""
+function build_compute_graph(tnc::TensorNetworkCircuit,
+                            plan::Vector{<:Tuple},
+                            bond_groups::Union{Nothing, Vector{<:Vector{<:Index}}}=nothing)
+    tnc = copy(tnc)
+    tn = tnc.tn
+    # create load and outputs nodes
+    nodes = Dict{Symbol, ComputeNode}()
+    tc = TensorCache()
+
+    # add a load node for each tensor
+    for t in keys(tnc)
+        data_symbol = push!(tc, tensor_data(tnc, t))
+        op = LoadCommand(t, data_symbol, collect(size(tnc[t])))
+        nodes[t] = ComputeNode{LoadCommand}(op)
+    end
+
+    # overwrite tensors corresponding to output tensors with output comamnds
+    for (i, o) in enumerate(output_tensors(tnc))
+        op = OutputCommand(o, i, size(tnc[o])[1])
+        nodes[o] = ComputeNode{OutputCommand}(op)
+    end
+
+    changed_ids = Dict{Symbol, Symbol}()
+    # add view commands to slice tensors
+    if bond_groups !== nothing
+        for (i, bg) in enumerate(bond_groups)
+            related_tensors = union([tnc[b] for b in bg]...)
+            slice_sym = Symbol("v$(i)")
+            slice_dim = QXTns.dim(bg[1])
+            for t in related_tensors
+                new_sym = Symbol("$(t)_s")
+                indices = hyperindices(tnc, t, all_indices=true)
+                # find which group overlaps with slice_bonds
+                index_position = findfirst(x -> length(intersect(bg, x)) > 0, indices)
+                op = ViewCommand(new_sym, t, slice_sym, index_position, slice_dim)
+                replace_tensor_symbol!(tn, t, new_sym)
+                changed_ids[t] = new_sym
+                node = ComputeNode{ViewCommand}(op)
```

Comment:
```suggestion
                node = ComputeNode(op)
```
I think the constructor for ComputeNodes allows you to do this without specifying the type